### PR TITLE
Feature/gal 857

### PR DESCRIPTION
**Pull request by chenglawson1**
## Description
https://nhsd-jira.digital.nhs.uk/browse/GAL-857

Written code to generate a signed JWT using private key created for the Comms Manager integration environment, and call Comms Manager API token endpoint to exchange for a bearer token.

Bearer token will be used to call Comms Manager API message endpoint to send a message to a single NHS number (this is to be done).

## Context

Bearer token required to call Comms Manager's application-restricted API.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by zainm97 — APPROVED** [review]
Looks good for interim code. Just a couple stuff we will need to change:
- use ESModules and imports instead of CommonJS requires (updating package.json with modules)
- On a previous ticket we had issues when env variabled were hard coded as function parameters, so maybe generalise functions with variables

File: terraform/src/sendSingleMessageToParticipantLambda/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@aws-sdk/client-secrets-manager": "^3.491.0",
```
Comment (Abdul-NHS) [inline terraform/src/sendSingleMessageToParticipantLambda/package.json:12]: Do we need to remove the dependencies that relate to aws for lambdas?

**Review by Abdul-NHS — APPROVED** [review]

File: terraform/src/sendSingleMessageToParticipantLambda/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "jti": uuidv4(),
+    "aud": tokenEndpointUrl,
+    "exp":  expiration,/* 5mins in the future */
+  };
+
+  const signedJWT = jwt.sign(claims, privateKey, { algorithm: "RS512", header: { kid: publicKeyId } });
+  return signedJWT;
+};
+
+const getAccessToken = async (tokenEndpointUrl, signedJWT) => {
```
Comment (avi312NHS) [inline terraform/src/sendSingleMessageToParticipantLambda/index.js:48]: probably need try catch here so we can handle any errors in the async data fetching

File: terraform/src/sendSingleMessageToParticipantLambda/index.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    },
+    data: qs.stringify(data)
+  };
+
+  const response = await axios(config)
+  console.log(response);
+  if (response.status === 200) {
+    return response.data;
+  } else {
+    return undefined;
```
Comment (avi312NHS) [inline terraform/src/sendSingleMessageToParticipantLambda/index.js:69]: can probably remove return undefined, and move this to catch block with an error message?

File: terraform/src/sendSingleMessageToParticipantLambda/index.js (hunk 2 above)
Comment (chenglawson1) [reply terraform/src/sendSingleMessageToParticipantLambda/index.js:48]: Any error is caught in the catch block in the handler.

File: terraform/src/sendSingleMessageToParticipantLambda/index.js (hunk 3 above)
Comment (chenglawson1) [reply terraform/src/sendSingleMessageToParticipantLambda/index.js:69]: Errors are caught in the handler catch block, tested scenario. The return in the else block is just for completeness.

**Review by avi312NHS — APPROVED** [review]

File: terraform/src/sendSingleMessageToParticipantLambda/package.json (hunk 1 above)
Comment (chenglawson1) [reply terraform/src/sendSingleMessageToParticipantLambda/package.json:12]: AWS sdk does not need to be in package.json, it is loaded into lambda by AWS, but no issue with having it in package.json either.

**Pull request merged**